industrial_ci/src/env.sh

```diff
@@ -56,22 +56,44 @@ if [ ! "$ROS_REPOSITORY_PATH" ]; then
     esac
 fi
 
-export UBUNTU_OS_CODE_NAME
-if [ -z "$UBUNTU_OS_CODE_NAME" ]; then
-    case "$ROS_DISTRO" in
-    "hydro")
-        UBUNTU_OS_CODE_NAME="precise"
-        ;;
-    "indigo"|"jade")
-        UBUNTU_OS_CODE_NAME="trusty"
-        ;;
-    "kinetic"|"lunar")
-        UBUNTU_OS_CODE_NAME="xenial"
-        ;;
-    *)
-        error "ROS distro '$ROS_DISTRO' is not supported"
-        ;;
-esac
+export OS_CODE_NAME
+export OS_NAME
+export DOCKER_BASE_IMAGE
+
+if [ -n "$OS_NAME" ] && [ -z "$OS_CODE_NAME" ]; then
+    error "please specify OS_CODE_NAME"
 fi
 
+if [ -n "$UBUNTU_OS_CODE_NAME" ]; then
+    OS_CODE_NAME=$UBUNTU_OS_CODE_NAME
+    OS_NAME=ubuntu
+fi
+
+if [ -n "$DOCKER_BASE_IMAGE" ]; then
+    OS_CODE_NAME=${OS_CODE_NAME:-${DOCKER_BASE_IMAGE##*:}} # use tag
+    OS_NAME=${OS_NAME:-${DOCKER_BASE_IMAGE%%:*}} # use repo
+else
+    if [ -n "$OS_NAME" ] && [ -z "$OS_CODE_NAME" ]; then
```

Comment:
Is this `if` clause intentionally repeated from [about 10 lines earlier](https://github.com/ros-industrial/industrial_ci/pull/174/files?diff=split#diff-3435d065f40f17428d8fba2554a30fdeR63)? If so why?